utils/set/sampleable_set.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package set
+
+import (
+	"bytes"
+
+	stdjson "encoding/json"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/json"
+	"github.com/ava-labs/avalanchego/utils/math"
+	"github.com/ava-labs/avalanchego/utils/sampler"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+)
+
+var _ stdjson.Marshaler = (*Set[int])(nil)
+
+// SampleableSet is a set of elements that supports sampling.
+type SampleableSet[T comparable] struct {
+	indicies map[T]int
```

Comment:
can we add a comment for this?
```suggestion
	indices map[T]int
```